runtime/executor/src/main/java/edu/snu/nemo/runtime/executor/datatransfer/InputReader.java

```diff
@@ -144,6 +145,21 @@ public RuntimeEdge getRuntimeEdge() {
     return runtimeEdge;
   }
 
+  /**
+   * Get block id.
+   *
+   * @param  taskIdx task index of the block
+   * @return the block id
+   */
+  private String getBlockId(final int taskIdx) {
+    final Pair<Integer, String> invariantDataProperty =
+        (Pair<Integer, String>) runtimeEdge.getProperty(ExecutionProperty.Key.InvariantData);
+    if (invariantDataProperty == null || invariantDataProperty.left() <= 1) {
```

Comment:
```java
int numOfDuplicateEdges = invariantDataProperty.left()
if () {
...
}
```

To make it explicit what `invariantDataProperty.left()` means.

Reply:
Because `invariantDataProperty` is nullable, getting value of the pair before checking if it is null is dangerous, so I fixed my code only where the `invariantDataProperty` is definitely not null.